src/test_util/components.rs

```diff
@@ -164,6 +180,119 @@ impl ComponentTester {
     }
 }
 
+/// Convenience wrapper for running source tests
+pub async fn assert_source_compliance<F, T>(tags: &[&str], f: F) -> T
+where
+    F: Future<Output = T>,
+{
+    init_test();
+
+    let result = f.await;
+
+    SOURCE_TESTS.assert(tags);
+
+    result
+}
+
+pub async fn run_and_assert_source_compliance<SC>(
+    source: SC,
+    timeout: Duration,
+    tags: &[&str],
+) -> Vec<Event>
+where
+    SC: SourceConfig,
+{
+    run_and_assert_source_compliance_with_setup(source, |_| {}, timeout, tags).await
+}
+
+pub async fn run_and_assert_source_compliance_with_setup<SC, F>(
```

Comment:
Could this be merged with `run_and_assert_source_compliance_n` below, where one uses an infinite `n` and the other an infinite `timeout`?

Reply:
Yeah, good call.

Now `run_and_assert_source_compliance_with_setup` is `run_and_assert_source_compliance_advanced` and both `run_and_assert_source_compliance` and `run_and_assert_source_compliance_n` forward to it, with your mutually exclusive infinites suggestion.